frontend/src/visualization/filtering/FilterLine.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+import { IconButton } from '@material-ui/core';
+import TuneIcon from '@material-ui/icons/Tune';
+import AddIcon from '@material-ui/icons/Add';
+import { from } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import FilterLineProperties from './FilterLineProperties';
+import useService from '../../dependency-injection/useService';
+import { FilterService } from '../../services/filter';
+import { EdgeTypeFilterModel, NodeTypeFilterModel } from '../../shared/filter';
+import FilterQueryStore from '../../stores/FilterQueryStore';
+import FilterStateStore from '../../stores/filterState/FilterStateStore';
+import useObservable from '../../utils/useObservable';
+import withLoadingBar from '../../utils/withLoadingBar';
+import withErrorHandler from '../../utils/withErrorHandler';
+import LoadingStore from '../../stores/LoadingStore';
+import ErrorStore from '../../stores/ErrorStore';
+
+type EntityTypeFilterModel = NodeTypeFilterModel | EdgeTypeFilterModel;
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    root: {
+      maxWidth: 150,
+      boxShadow: 'none',
+      textTransform: 'none',
+      fontSize: 16,
+      padding: '6px 12px',
+      border: '1px solid',
+      lineHeight: 1.5,
+      '&:hover': {
+        boxShadow: 'none',
+      },
+    },
+  })
+);
+
+const FilterLine = (props: {
+  backgroundColor: string;
+  type: string;
+  entity: 'node' | 'edge';
+}): JSX.Element => {
+  const classes = useStyles();
+  const { backgroundColor, type, entity } = props;
+
+  // Indicates if filter-dialog is opened.
+  const [filterOpen, setFilterOpen] = React.useState(false);
+  const [boxShadow, setBoxShadow] = React.useState('None');
+  const isActive = React.useRef(false);
+
+  const filterService = useService(FilterService, null);
+
+  const filterQueryStore = useService<FilterQueryStore>(FilterQueryStore);
+  const filterStateStore = useService<FilterStateStore>(FilterStateStore);
+
+  const loadingStore = useService<LoadingStore>(LoadingStore);
+  const errorStore = useService<ErrorStore>(ErrorStore);
+
+  const [filterModel, setFilterModel] = useState<EntityTypeFilterModel>({
+    name: '',
+    properties: [],
+  });
+
+  useObservable(
```

Comment:
take the state that you get from `useObservable`. No need for the extra `useState<EntityTypeFilterModel`